.github/workflows/e2e-test-linux-vm.yaml

```
@@ -5,47 +5,73 @@ on:
+          docker-image-tag: ${{env.IMAGE_VERSION}}
+
+      - name: Install gha-runner-scale-set-controller
+        id: install_arc_controller
         run: |
-            DOCKER_CLI_EXPERIMENTAL=enabled DOCKER_BUILDKIT=1 docker buildx build --build-arg RUNNER_VERSION=$RUNNER_VERSION --build-arg TAG=$TAG -t $IMAGE . --load
-      - name: Create Kind cluster
+          helm install arc \
+          --namespace "arc-systems" \
+          --create-namespace \
+          --set image.repository=${{ env.IMAGE_NAME }} \
+          --set image.tag=${{ env.IMAGE_VERSION }} \
+          ./charts/gha-runner-scale-set-controller \
+          --debug
+          sleep 30
+          kubectl get pod -n arc-systems
+          kubectl describe deployment arc-gha-runner-scale-set-controller -n arc-systems
+
+      - name: Install gha-runner-scale-set
+        id: install_arc
         run: |
-          PATH=$(go env GOPATH)/bin:$PATH
-          kind create cluster --name $CLUSTER_NAME
-      - name: Load Image to Kind Cluster
-        run: kind load docker-image $IMAGE --name $CLUSTER_NAME
-      - name: Get Token
-        id: get_workflow_token
-        uses: peter-murray/workflow-application-token-action@8e1ba3bf1619726336414f1014e37f17fbadf1db
-        with:
-          application_id: ${{ secrets.ACTIONS_ACCESS_APP_ID }}
-          application_private_key: ${{ secrets.ACTIONS_ACCESS_PK }}
-          organization: ${{ env.TARGET_ORG }}
-      - uses: ./.github/actions/e2e-arc-test
-        with:
-          github-token: ${{ steps.get_workflow_token.outputs.token }}
-          config-url: "https://github.com/actions-runner-controller/arc_e2e_test_dummy"
-          docker-image-repo: $IMAGE_REPO
-          docker-image-tag: $TAG
+          ARC_NAME=arc-runner-${{github.job}}-$(date +'%M-%S')-$(($RANDOM % 100 + 1))
+          helm install "$ARC_NAME" \
+          --namespace "arc-runners" \
+          --create-namespace \
+          --set githubConfigUrl="${{ env.TEST_CONFIG_URL }}" \
+          --set githubConfigSecret.github_token="${{ steps.setup.outputs.token }}" \
+          ./charts/gha-runner-scale-set \
+          --debug
+          echo "ARC_NAME=$ARC_NAME" >> $GITHUB_OUTPUT
+          sleep 30
+          kubectl get pod -n arc-systems
+
+      - name: Test ARC scales pods up and down
+        run: |
+          export GITHUB_TOKEN="${{ steps.setup.outputs.token }}"
+          export ARC_NAME="${{ steps.install_arc.outputs.ARC_NAME }}"
+          go test ./test_e2e_arc -v
+
+      - name: Uninstall gha-runner-scale-set
+        if: steps.install_arc.outcome == 'success'
+        run: |
+          helm uninstall ${{ steps.install_arc.outputs.ARC_NAME }} --namespace arc-runners
+          sleep 30
+
+      - name: Dump gha-runner-scale-set-controller logs
+        if: steps.install_arc_controller.outcome == 'success'
+        run: |
+          kubectl logs deployment/arc-gha-runner-scale-set-controller -n arc-systems
```

```suggestion
          kubectl logs deployments/arc-gha-runner-scale-set-controller -n arc-systems
```
I think it should be deployments, this might cause error later

---

it was working fine with my syntax. 😄 
https://github.com/actions/actions-runner-controller/actions/runs/4395784129/jobs/7697812090#step:8:2

---

Yeah, that one is confusing me
Because kubectl autocomplete completes to deployments :relaxed: